Taarafo.Core.Tests.Unit/Services/Foundations/Comments/CommentServiceTests.Validation.Modify.cs

```diff
@@ -369,5 +370,67 @@ public async Task ShouldThrowValidationExceptionOnModifyIfStorageUpdatedDateSame
             this.dateTimeBrokerMock.VerifyNoOtherCalls();
             this.loggingBrokerMock.VerifyNoOtherCalls();
         }
+
+        [Fact]
+        public async void ShouldThrowValidationExceptionOnModifyWhenReferneceExceptionAndLogItAsync()
+        {
+            // given
+            int minuteInPast = GetRandomNegativeNumber();
+            DateTimeOffset randomDate = GetRandomDateTimeOffset();
+            Comment randomComment = CreateRandomModifyComment(randomDate.AddMinutes(minuteInPast));
+            Comment inputComment = randomComment.DeepClone();
+            inputComment.UpdatedDate = randomDate;
+            Comment storageComment = randomComment;
+            string randomMessage = GetRandomMessage();
+            string exceptionMessage = randomMessage;
+            var foreignKeyConstraintConflictException = new ForeignKeyConstraintConflictException(exceptionMessage);
+
+            var invalidCommentReferenceException =
+                new InvalidCommentReferenceException(foreignKeyConstraintConflictException);
+
+            var expectedCommentValidationException =
+                new CommentValidationException(invalidCommentReferenceException);
```

Comment:
This will be dependency validation exception and goes under 'Exceptions' file

Reply:
@ShriHumrudha, I have updated this